src/Maestro/maestro-angular/src/app/page/build-graph-table/build-graph-table.component.ts

```diff
@@ -184,6 +189,59 @@ function buildHasIncoherentDependencies(buildData: BuildData, includeToolsets: b
   return false;
 }
 
+function hasCycles(build:Build, buildData: BuildData[]): [boolean,string?] {
+  const repository = getRepo(build)
+  let currentBuildData = buildData.find(r => r.build.id == build.id);
+  
+  // Check the current build to see if any of its product dependencies or subdependencies introduce
+  // a cycle with the current repository
+  if (currentBuildData && repository)
+  {
+    let [hasCycle, cyclePath] = dependencyHasCycle(currentBuildData,buildData, repository)
+    if (hasCycle)
+    {
+      let newCycle = [getRepo(currentBuildData.build), cyclePath];
+      return [true, newCycle.join('->\n')];
+    }
+    return dependencyHasCycle(currentBuildData, buildData, repository);
```

Comment:
You already called this same function higher up. Just return `[hasCycle, cyclePath]`